### PR TITLE
Add refresh button to the chat panel

### PR DESCRIPTION
**Pull request by manodnyab**
## Description

Refresh and start a new session in Q chat.
Reference: https://github.com/aws/aws-toolkit-jetbrains/pull/4651

https://github.com/user-attachments/assets/3f931c9c-668a-4e57-810f-a6de0ce8864d

 
## License
I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt [hunk 1]
```diff
@@ -44,11 +50,35 @@ class AmazonQToolWindowFactory : ToolWindowFactory, DumbAware {
...
+                            AmazonQToolWindow.getInstance(project).component
+                        } else {
+                            QWebviewPanel.getInstance(project).browser?.prepareBrowser(BrowserState(FeatureId.Q))
+                            QWebviewPanel.getInstance(project).component
+                        }
+
+                        val content = contentManager.factory.createContent(component, null, false).also {
+                            it.isCloseable = true
+                            it.isPinnable = true
+                        }
```
Comment (Will-ShaoHua) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt:69]: let's dedupe these lines https://github.com/aws/aws-toolkit-jetbrains/blob/main/plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt#L66-L76

Comment (manodnyab) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/toolwindow/AmazonQToolWindowFactory.kt]: Deduped!

**Review by Will-ShaoHua — APPROVED** [review]

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/webview/Browser.kt [hunk 2]
```diff
@@ -32,6 +33,10 @@ class Browser(parent: Disposable) {
         loadWebView(isCodeTransformAvailable, isFeatureDevAvailable)
     }
 
+    override fun dispose() {
+        Disposer.dispose(jcefBrowser)
+    }
+
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/webview/Browser.kt:39]: isnt it already parented to a disposable

Comment (Will-ShaoHua) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/webview/Browser.kt:39]: yea, the parent here is `AmazonQToolWindow`, but here we only want to dispose this jcefBrowser and its down-stream?

Comment (rli) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/webview/Browser.kt:39]: then shouldnt it be reparented and we can drop the parent param?

Comment (manodnyab) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/webview/Browser.kt:39]: So remove the parent param and while registering the disposer send the current browser instance?

Comment (Will-ShaoHua) [reply plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/webview/Browser.kt:39]: yea, we should just do 
```
val jcefBrowser = createBroowser(this)
```
and we can remove parent disposable from ctor

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QRefreshPanelAction.kt [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+import com.intellij.openapi.actionSystem.AnActionEvent
+import com.intellij.openapi.application.ApplicationManager
+import com.intellij.openapi.project.DumbAwareAction
+import com.intellij.util.messages.Topic
+import software.aws.toolkits.jetbrains.services.amazonq.toolwindow.AmazonQToolWindow
+import software.aws.toolkits.resources.AmazonQBundle
+import software.aws.toolkits.resources.message
+import java.util.EventListener
+
+class QRefreshPanelAction : DumbAwareAction(AmazonQBundle.message("amazonq.refresh.panel"), "abcs", AllIcons.Actions.Refresh) {
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QRefreshPanelAction.kt:17]: `abcs`

File: plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QRefreshPanelAction.kt [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+    override fun actionPerformed(e: AnActionEvent) {
+        val project = e.project ?: return
+        // recreate chat browser
+        AmazonQToolWindow.getInstance(project).disposeAndRecreate()
+        // recreate signin browser
+        QWebviewPanel.getInstance(project).disposeAndRecreate()
+        RefreshQChatPanelButtonPressedListener.notifyRefresh()
```
Comment (rli) [inline plugins/amazonq/chat/jetbrains-community/src/software/aws/toolkits/jetbrains/services/amazonq/QRefreshPanelAction.kt:24]: listeners are not the right pattern for a single sender -> single receiver
let's fix this later

File: plugins/amazonq/shared/jetbrains-community/resources/software/aws/toolkits/resources/AmazonQBundle.properties [hunk 5]
```diff
@@ -1,2 +1,3 @@
 action.q.hello.description=Hello description
+amazonq.refresh.panel=Refresh Chat Session
```
Comment (rli) [inline plugins/amazonq/shared/jetbrains-community/resources/software/aws/toolkits/resources/AmazonQBundle.properties:2]: i think the `action.<id>` syntax should have worked

**Review by rli — APPROVED** [review]

**Comment by rli** [comment]
windows OOM

**Pull request merged**